core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -106,6 +115,13 @@ public static UnderFileSystemFactory find(String path) {
         return factory;
       }
     }
+    for (UnderFileSystemFactory factory : scanExtensions()) {
```

Comment:
instead of having a second `for` loop, could we call `scanExtensions()` before the first `for` loop to populate `FACTORIES`?